### PR TITLE
Some bug fixes for the code generator

### PR DESCRIPTION
**Pull request by szeiger**
Review by @cvogt

### CODE REVIEW

File: slick-codegen/src/main/scala/scala/slick/codegen/AbstractGenerator.scala [hunk 1]
```diff
@@ -44,7 +44,7 @@ abstract class AbstractGenerator[Code,TermName,TypeName](model: m.Model)
     /** Column code generators in the order they appear in the model. */
     final lazy val columnsPositional: IndexedSeq[Column] = model.columns.map(Column).toIndexedSeq
     /** Database column positions in the desired user-facing order. Currently just moves the positions of AutoInc columns to the end if autoIncLastAsOption is enabled. */
-    final lazy val desiredColumnOrder: Seq[Int] = {
+    lazy val desiredColumnOrder: Seq[Int] = {
```
Comment (cvogt) [inline slick-codegen/src/main/scala/scala/slick/codegen/AbstractGenerator.scala]: What this change?

Comment (szeiger) [inline slick-codegen/src/main/scala/scala/slick/codegen/AbstractGenerator.scala]: Requested in #881. Seems like it should be a harmless change.

Comment (cvogt) [inline slick-codegen/src/main/scala/scala/slick/codegen/AbstractGenerator.scala]: I see. Would be nice if we could make this orthogonal to autoIncAsOption somehow. Not sure that's sensible.

**Comment by cvogt** [comment]
LGTM

**Pull request merged**